packages/components/src/components/post-language-switch/post-language-switch.tsx

```diff
@@ -59,24 +60,34 @@ export class PostLanguageSwitch {
     this.updateActiveLanguage();
   }
 
+  /**
+   * Listen for the postChange event and guard it to run only for events originating from a <post-language-option> element.
+   */
   @Listen('postChange')
   handlePostChange(event: CustomEvent<string>) {
-    this.activeLang = event.detail;
-
-    // Update the active state in the children post-language-option components
-    this.languageOptions.forEach(lang => {
-      if (lang.code && lang.code === this.activeLang) {
-        lang.setAttribute('active', '');
-      } else {
-        lang.removeAttribute('active');
+    eventGuard.call(
+      this,
+      event,
+      { targetLocalName: 'post-language-option' },
```

Comment:
I think here, we should specify the delegatorSelector with the value `post-language-swtich`.